docs/private-dns/api-reference.md

```diff
@@ -0,0 +1,561 @@
+##### Summary:
+
+Gets an existing device by ID
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | Device info |
+| 404 | Device not found |
+
+#### PUT
+##### Summary:
+
+Updates an existing device
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | Device updated |
+| 400 | Validation failed |
+| 404 | Device not found |
+
+### /oapi/v1/devices/{device_id}/doh.mobileconfig
+
+#### GET
+##### Summary:
+
+Get DNS over HTTPS .mobileconfig.
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+| exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |
+| exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | DNS over HTTPS plist file |
+| 404 | Device not found |
+
+### /oapi/v1/devices/{device_id}/dot.mobileconfig
+
+#### GET
+##### Summary:
+
+Get DNS over TLS .mobileconfig.
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+| exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |
+| exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | DNS over HTTPS plist file |
+| 404 | Device not found |
+
+### /oapi/v1/devices/{device_id}/settings
+
+#### PUT
+##### Summary:
+
+Updates device settings
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | Device settings updated |
+| 400 | Validation failed |
+| 404 | Device not found |
+
+### /oapi/v1/dns_servers
+
+#### GET
+##### Summary:
+
+Lists DNS servers
+
+##### Description:
+
+Lists DNS servers for user. By default exists default DNS server
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | List of DNS servers |
+
+#### POST
+##### Summary:
+
+Creates a new DNS server
+
+##### Description:
+
+Creates a new DNS server. You can attach custom settings, otherwise DNS server will be created with default settings.
```

Comment:
```suggestion
Creates a new DNS serve that you can customize to your liking.
```
or, more formally:
```suggestion
Creates a new DNS serve that you can customize according to your preferences.
```

Reply:
Not really, the current description is more precise which is important for an API document.